ocaml/xapi/message_forwarding.ml

```diff
@@ -6464,4 +6472,183 @@ functor
               ~base_build_id ~base_version ~base_release ~to_version ~to_release
         )
     end
+
+    module Tracing = struct
+      let do_trace_op_on ~__context ~self ~local_fn ~client_fn ~db_fn =
+        let localhost = Helpers.get_localhost ~__context in
+        if Helpers.is_pool_master ~__context ~host:localhost then (
+          let hosts =
+            match Db.Tracing.get_hosts ~__context ~self with
+            | [] ->
+                Db.Host.get_all ~__context
+            | hosts ->
+                hosts
+          in
+          let hosts, local = List.partition (( <> ) localhost) hosts in
+          List.iter
+            (fun host -> do_op_on ~__context ~local_fn ~host client_fn)
+            hosts ;
+          List.iter (fun _ -> local_fn ~__context) local ;
```

Comment:
A small name change makes the reason of the enforced order more obious:
```suggestion
          let members, coordinator = List.partition (( <> ) localhost) hosts in
          List.iter
            (fun host -> do_op_on ~__context ~local_fn ~host client_fn)
            members ;
          List.iter (fun _ -> local_fn ~__context) coordinator ;
```

Reply:
Resolved [2eab70b](https://github.com/xapi-project/xen-api/pull/4966/commits/2eab70b666d23cd0747c1fc823356685c1b86e93)